### DISCUSSION TITLE
applySnapshot when the model is using a preprocessor

### DISCUSSION THREAD

**Discussion by aqrojo** [mobxjs/mobx-state-tree#1736] [Q&A] [answered]
I have a model that uses a preprocessor to initialize its state and it isn't re-hydrating when I try to apply a snapshot
[Here is an example](https://codesandbox.io/embed/autumn-rain-nxt61?fontsize=14&hidenavigation=1&theme=dark) 
Maybe I'm missing something?

**Comment by jamonholmgren** [comment, marked as answer]
@aqrojo You have to map the values the same way you did the first time. Here's how that works:

https://codesandbox.io/s/optimistic-borg-z79gm?file=/src/index.js

![image](https://user-images.githubusercontent.com/1479215/125365467-2b9ff680-e329-11eb-9021-8afdb52400a2.png)

**Reply by aqrojo** [reply]
It works perfectly, thanks so much for the help!
